x-pack/legacy/plugins/uptime/public/state/api/snapshot.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ThrowReporter } from 'io-ts/lib/ThrowReporter';
+import { isRight } from 'fp-ts/lib/Either';
+import { getApiPath } from '../../lib/helper';
+import { SnapshotType, Snapshot } from '../../../common/runtime_types';
+
+interface ApiRequest {
+  basePath: string;
+  dateRangeStart: string;
+  dateRangeEnd: string;
+  filters?: string;
+  statusFilter?: string;
+}
+
+export const fetchSnapshotCount = async ({
+  basePath,
+  dateRangeStart,
+  dateRangeEnd,
+  filters,
+  statusFilter,
+}: ApiRequest): Promise<Snapshot> => {
+  const url = getApiPath(`/api/uptime/snapshot/count`, basePath);
+  const urlParams = encodeURI(
+    `?dateRangeStart=${dateRangeStart}&dateRangeEnd=${dateRangeEnd}${
+      filters ? `&filters=${filters}` : ''
+    }${statusFilter ? `&statusFilter=${statusFilter}` : ''}`
+  );
```

Comment:
i wonder if it's worth using URLSearchParams function here

  ```
const params = {
    dateRangeStart,
    dateRangeEnd,
    ...(filters && { filters }),
    ...(statusFilter && { statusFilter }),
  };
  const urlParams = new URLSearchParams(params).toString();

```

Reply:
Does it looks more clean , though looks more complex 🍡

Reply:
Definitely looks cleaner, which is usually my preference

Reply:
Resolved in https://github.com/elastic/kibana/pull/48035/commits/887d9fb63c5fe8c21a3cbcb8d94939bf23e7d5b7.